chia/wallet/vc_wallet/vc_store.py

```diff
@@ -250,9 +250,16 @@ async def get_vc_record_by_coin_id(self, coin_id: bytes32) -> Optional[VCRecord]
 
     async def add_vc_proofs(self, vc_proofs: VCProofs) -> None:
         async with self.db_wrapper.writer_maybe_transaction() as conn:
-            await conn.execute(
-                "INSERT INTO vc_proofs VALUES(?, ?)", (vc_proofs.root().hex(), bytes(vc_proofs.as_program()))
-            )
+            root: bytes32 = vc_proofs.root()
+            exists_cursor = await conn.execute("SELECT EXISTS(SELECT 1 FROM vc_proofs WHERE root=?)", (root.hex(),))
+            root_exists = await exists_cursor.fetchone()
+            await exists_cursor.close()
+            if root_exists is not None and root_exists[0] != 0:
+                existing_proof = await self.get_proofs_for_root(root)
+                assert existing_proof is not None
+                if existing_proof == vc_proofs:
```

Comment:
as far as I can tell, the UNIQUE constraint only applies to the `root` (since it's the `PRIMARY KEY`). So if this comparison fails, you'll still try to insert the new proof and get the DB constraint violation.

If this is deliberate, it would be good to have a test for it.

Or perhaps the mistake is in the table constraints, perhaps the *pair* of `root` and `proof` is supposed to be unique.